libs/remix-ui/run-tab/src/lib/actions/events.ts

```diff
@@ -94,22 +96,7 @@ export const setupEvents = (plugin: RunTab, dispatch: React.Dispatch<any>) => {
   plugin.on('filePanel', 'setWorkspace', async () => {
     dispatch(resetUdapp())
     resetAndInit(plugin)
-    const { network } = await plugin.call('blockchain', 'getCurrentNetworkStatus')
-    const dirName = plugin.REACT_API.networkName === 'VM' ? plugin.REACT_API.selectExEnv : network.id
-    const isPinnedAvailable = await plugin.call('fileManager', 'exists', `.deploys/pinned-contracts/${dirName}`)
-    if (isPinnedAvailable) {
-      try {
-        const list = await plugin.call('fileManager', 'readdir', `.deploys/pinned-contracts/${dirName}`)
-        const filePaths = Object.keys(list)
-        for (const file of filePaths) {
-          const pinnedContract = await plugin.call('fileManager', 'readFile', file)
-          const pinnedContractObj = JSON.parse(pinnedContract)
-          if (pinnedContractObj) addPinnedInstance(dispatch, pinnedContractObj)
-        }
-      } catch(err) {
-        console.log(err)
-      }
-    }
+    await migrateSavedContracts(plugin)
```

Comment:
on the `setWorkspace` you don't load pinned contracts?

Reply:
yes, we load it once network is ready

Reply:
yes but when the workspace change you still need to reload pinned contract no?

Reply:
when workspace is changed, network is also reset/updated